### PR TITLE
Convert Many things to Use Brewcask

### PR DESCRIPTION
**Pull request by lyzadanger**
This is a PR that touches a lot of things but is working on one primary theme: where possible, **relying on the `brewcask` provider to manage binary software where before we were using either standalone `boxen-puppet` modules or custom modules we built ourselves.**

As such, what you'll see a lot of in this diff is RED :red_circle: . The goal here is a reduction in complexity and maintenance burden. The `brewcask` maintenance community is vastly larger than that of specialized puppet-boxen modules, so I want to benefit from that.

Managing binaries with `brewcask` is much simpler than adding a module. Much like how we install things with `brew`, it's a one-liner:

`package { 'atom': provider => 'brewcask' }`

You can find available `brewcasks` on [the brewcask site](http://caskroom.io/search).
## Removed Dependencies on Puppet-Boxen Modules

These items were converted to use `brewcask` from previous standalone modules:
- chrome
- firefox
- fluid
- harvest
- iterm2
- phantomjs
- skype
## Removed a few Self-Made Custom Modules

The following were previously managed with custom modules:
- atom
- licecap

I also removed the `sublime-text` custom modules we were maintaining. The next new employee we have who wants sublime text will prompt me to find a good `brewcask` or other simple-ish solution at that time.
## Added Dropbox Back

The puppet-boxen dropbox module wasn't being maintained and the binary referenced was woefully out of date. Fixed by using `brewcask` provider instead.
## Updated People Manifests

I updated everyone's individual manifest as needed to switch things over to brewcask. Because the software involved is _already installed on those users' machines_, these changes should have no visible effect. More details are available in comments in those manifests and the commit history here. 
## In Summary

These changes are intended to simplify our use of `boxen` and reduce maintenance concern over time.

@TODO: Update the wiki with `brewcask` recipes and whatnot after merge.

### CODE REVIEW

**Comment by tylersticka** [comment]
This makes sense to me, @lyzadanger.

I just tried pulling this branch and running `boxen`, but I got a _bunch_ of "download failed" errors. Not sure if this is a failure of the repo or a failure of these changes.

Full error output below:

```
~/src/our-boxen feature-use-brewcask
❯ boxen
--> Preparing to auto-update...
--> Skipping auto-update, because your boxen checkout isn't on master

Working on something special, over here on a branch? That's cool! Once
you're done, you'll probably want to submit a pull request and get back
onto master so that you can get some sweet, sweet updates again, though.

For more info on how the simple git-based updates work, have a gander at
https://github.com/boxen/our-boxen/blob/master/docs/updates.md

Fact file /opt/boxen/repo/facts.d/example.yaml was parsed but returned an empty data set
Notice: Scope(Class[People::Tylersticka]): Hi, Tyler! Happy designing 😄
Notice: Compiled catalog for discovery-one.local in environment production in 3.92 seconds
Notice: /Stage[main]/People::Tylersticka/Exec[link-dotfiles]/returns: executed successfully
Error: Execution of 'brew cask install dropbox' returned 1: ==> Downloading https://www.dropbox.com/download?plat=mac&full=1
Error: Download failed on Cask 'dropbox' with message: Download failed: https://www.dropbox.com/download?plat=mac&full=1
Error: /Stage[main]/Cloudfour/Package[dropbox]/ensure: change from absent to present failed: Execution of 'brew cask install dropbox' returned 1: ==> Downloading https://www.dropbox.com/download?plat=mac&full=1
Error: Download failed on Cask 'dropbox' with message: Download failed: https://www.dropbox.com/download?plat=mac&full=1
Error: Execution of 'brew cask install firefox' returned 1: ==> Downloading https://ftp.mozilla.org/pub/firefox/releases/41.0.2/mac/en-US/Firefox%2041.0.2.dmg
Error: Download failed on Cask 'firefox' with message: Download failed: https://ftp.mozilla.org/pub/firefox/releases/41.0.2/mac/en-US/Firefox%2041.0.2.dmg
Error: /Stage[main]/Cloudfour/Package[firefox]/ensure: change from absent to present failed: Execution of 'brew cask install firefox' returned 1: ==> Downloading https://ftp.mozilla.org/pub/firefox/releases/41.0.2/mac/en-US/Firefox%2041.0.2.dmg
Error: Download failed on Cask 'firefox' with message: Download failed: https://ftp.mozilla.org/pub/firefox/releases/41.0.2/mac/en-US/Firefox%2041.0.2.dmg
Error: Execution of 'brew cask install licecap' returned 1: ==> Downloading http://www.cockos.com/licecap/licecap125.dmg
Error: Download failed on Cask 'licecap' with message: Download failed: http://www.cockos.com/licecap/licecap125.dmg
Error: /Stage[main]/People::Tylersticka/Package[licecap]/ensure: change from absent to present failed: Execution of 'brew cask install licecap' returned 1: ==> Downloading http://www.cockos.com/licecap/licecap125.dmg
Error: Download failed on Cask 'licecap' with message: Download failed: http://www.cockos.com/licecap/licecap125.dmg
Error: Execution of 'brew cask install google-chrome' returned 1: ==> Caveats
The Mac App Store version of 1Password won't work with a Homebrew-Cask-linked Google Chrome. To bypass this limitation, you need to either:

  + Move Google Chrome to your /Applications directory (the app itself, not a symlink).
  + Install 1Password from outside the Mac App Store (licenses should transfer automatically, but you should contact AgileBits about it).

==> Downloading https://dl.google.com/chrome/mac/stable/GGRO/googlechrome.dmg
Error: Download failed on Cask 'google-chrome' with message: Download failed: https://dl.google.com/chrome/mac/stable/GGRO/googlechrome.dmg
Error: /Stage[main]/Cloudfour/Package[google-chrome]/ensure: change from absent to present failed: Execution of 'brew cask install google-chrome' returned 1: ==> Caveats
The Mac App Store version of 1Password won't work with a Homebrew-Cask-linked Google Chrome. To bypass this limitation, you need to either:

  + Move Google Chrome to your /Applications directory (the app itself, not a symlink).
  + Install 1Password from outside the Mac App Store (licenses should transfer automatically, but you should contact AgileBits about it).

==> Downloading https://dl.google.com/chrome/mac/stable/GGRO/googlechrome.dmg
Error: Download failed on Cask 'google-chrome' with message: Download failed: https://dl.google.com/chrome/mac/stable/GGRO/googlechrome.dmg
Error: Execution of 'brew cask install skype' returned 1: ==> Downloading https://www.skype.com/go/getskype-macosx.dmg
Error: Download failed on Cask 'skype' with message: Download failed: https://www.skype.com/go/getskype-macosx.dmg
Error: /Stage[main]/Cloudfour/Package[skype]/ensure: change from absent to present failed: Execution of 'brew cask install skype' returned 1: ==> Downloading https://www.skype.com/go/getskype-macosx.dmg
Error: Download failed on Cask 'skype' with message: Download failed: https://www.skype.com/go/getskype-macosx.dmg
Error: Execution of 'brew cask install harvest' returned 1: ==> Downloading https://www.getharvest.com/harvest/mac/Harvest.1.4.4.zip
Error: Download failed on Cask 'harvest' with message: Download failed: https://www.getharvest.com/harvest/mac/Harvest.1.4.4.zip
Error: /Stage[main]/Cloudfour/Package[harvest]/ensure: change from absent to present failed: Execution of 'brew cask install harvest' returned 1: ==> Downloading https://www.getharvest.com/harvest/mac/Harvest.1.4.4.zip
Error: Download failed on Cask 'harvest' with message: Download failed: https://www.getharvest.com/harvest/mac/Harvest.1.4.4.zip
```

**Comment by lyzadanger** [comment]
@tylersticka Thanks for being a brave explorer! It _looks_ like all but one of those errors is a download issue, but given that the downloads are from all sorts of different remotes, that's odd.

There is one error that may be legit to something I didn't think of:

```
Error: Download failed on Cask 'google-chrome' with message: Download failed: https://dl.google.com/chrome/mac/stable/GGRO/googlechrome.dmg
Error: /Stage[main]/Cloudfour/Package[google-chrome]/ensure: change from absent to present failed: Execution of 'brew cask install google-chrome' returned 1: ==> Caveats
The Mac App Store version of 1Password won't work with a Homebrew-Cask-linked Google Chrome. To bypass this limitation, you need to either:

  + Move Google Chrome to your /Applications directory (the app itself, not a symlink).
  + Install 1Password from outside the Mac App Store (licenses should transfer automatically, but you should contact AgileBits about it).
```

The `brewcask`-ed Chrome might not play friendly with 1Password from the App Store. This is a _touch_ odd as _I also_ have an AppStore version of 1Password and it didn't give me grief. Huh.

More soon...

**Comment by lyzadanger** [comment]
Bah: https://github.com/boxen/puppet-brewcask/issues/45

(more soon...)

**Comment by lyzadanger** [comment]
p.s. The sudoers hack referenced in that issue introduced _other_ problems for me when I tried it out yesterday in the context of getting `virtualbox` to install.

**Comment by lyzadanger** [comment]
@tylersticka Could you run again to see if you get the same failures?
